### PR TITLE
Fire focus event even when window is not in focus

### LINKED ISSUE #3: Question: Should we use Chrome on Travis?
Should we use Chrome on Travis instead of PhantomJS?

Looks like master is failing using PhantomJS, https://travis-ci.org/cibernox/ember-native-dom-helpers/branches

Idea: http://blog.500tech.com/setting-up-travis-ci-to-run-tests-on-latest-google-chrome-version/

### PR DESCRIPTION
**Pull request by pixelhandler**
- When running tests window may not be in focus, so manually fire focus, otherwise use `el.focus()`
- Remove `$` (jQuery) from helper
- Use real browsers, not just PhantomJS

Fixes #3

### CODE REVIEW

File: tests/integration/key-event-test.js [hunk 1]
```diff
@@ -34,12 +34,16 @@ test('It can fire keyup events', function(assert) {
...
   this.onKeyPress = (e) => {
     assert.ok(true, 'a focus event is fired');
     assert.ok(e instanceof window.Event, 'It receives a native event');
-    assert.equal(e.keyCode, 40, 'The event has the right keyCode');
-    assert.equal(e.which, 40, 'The event has the right which');
+    // event.which is deprecated, favor `keyCode`
+    if (e.keyCode) {
+      assert.equal(e.keyCode, 40, 'The event has the right keyCode');
+    } else {
+      assert.equal(e.which, 40, 'The event has the right which');
```
Comment (pixelhandler) [inline tests/integration/key-event-test.js]: This check for `e.which` fails in Firefox, so only check conditionally

File: addon-test-support/helpers.js [hunk 2]
```diff
@@ -1,27 +1,27 @@
...
       run(null, function() {
         // Firefox does not trigger the `focusin` event if the window
-        // does not have focus. If the document doesn't have focus just
-        // use trigger('focusin') instead.
-
-        if (!document.hasFocus || document.hasFocus()) {
-          el.focus();
+        // does not have focus. If the document does not have focus then
+        // fire `focusin` event as well.
+        if (document.hasFocus && !document.hasFocus()) {
```
Comment (pixelhandler) [inline addon-test-support/helpers.js]: The gotcha with focus is that Firefox won't fire `focus` if the window isn't in focus. (i.e. launched in background with testem). So just fire the event anyway.

File: addon-test-support/helpers.js [hunk 3]
```diff
@@ -53,9 +53,9 @@ function fireEvent(element, type, options = {}) {
   element.dispatchEvent(event);
 }
 
-function buildBasicEvent(type, options = {}) {
+function buildBasicEvent(type, options = {}, bubbles = true, cancelable = true) {
   let event = document.createEvent('Events');
-  event.initEvent(type, true, true);
+  event.initEvent(type, bubbles, cancelable);
```
Comment (pixelhandler) [inline addon-test-support/helpers.js]: Needed a way to fire a focus event when the document (window) doesn't have focus. The focus event is does not 'bubble'

File: testem.js [hunk 4]
```diff
@@ -3,10 +3,12 @@ module.exports = {
   "test_page": "tests/index.html?hidepassed",
   "disable_watching": true,
   "launch_in_ci": [
-    "PhantomJS"
+    "PhantomJS",
+    "Firefox",
+    "Chrome"
   ],
   "launch_in_dev": [
-    "PhantomJS",
```
Comment (pixelhandler) [inline testem.js]: Maybe in the future remove PhantomJS altogether, but no need in dev anyhow.

**Comment by pixelhandler** [comment]
@cibernox @rwjblue can someone merge this?

File: addon-test-support/helpers.js (hunk 2 above)
Comment (pixelhandler) [inline addon-test-support/helpers.js]: @rwjblue I'd like to propose the same change to https://github.com/emberjs/ember.js/blob/aeeddf7c00b42a340494465252481b5414d87009/packages/ember-testing/lib/events.js#L8-L27 as Firefox can fail to use focus when the window is not in focus.

File: .travis.yml [hunk 5]
```diff
@@ -26,13 +30,19 @@ matrix:
 
 before_install:
   - npm config set spin false
-  - npm install -g bower phantomjs-prebuilt
+  - npm install -g bower phantomjs-prebuilt yarn
```
Comment (cibernox) [inline .travis.yml]: Yarn recommends to install it from a script instead from npm. Apparently there was some edge cases with it. https://yarnpkg.com/lang/en/docs/install-ci/#travis-tab

File: addon-test-support/helpers.js [hunk 6]
```diff
@@ -1,27 +1,27 @@
...
 const KEYBOARD_EVENT_TYPES = ['keydown', 'keypress', 'keyup'];
 const MOUSE_EVENT_TYPES = ['click', 'mousedown', 'mouseup', 'dblclick', 'mouseenter', 'mouseleave', 'mousemove', 'mouseout', 'mouseover'];
 
 function focus(el) {
   if (!el) { return; }
-  let $el = $(el);
-  if ($el.is(':input, [contenteditable=true]')) {
-    let type = $el.prop('type');
+
+  if (el.tagName === 'INPUT' || el.contentEditable || el.tagName === 'ANCHOR') {
```
Comment (cibernox) [inline addon-test-support/helpers.js]: `ANCHOR`? If this is for `<a>` tags the tagName is `A`

File: tests/integration/key-event-test.js (hunk 1 above)
Comment (pixelhandler) [inline tests/integration/key-event-test.js]: @cibernox I can try that

File: .travis.yml (hunk 5 above)
Comment (pixelhandler) [inline .travis.yml]: I tried about 10 ways to run yarn and found some of the travis docs to not Just Work™ I did attempt two variations of an apt-get. Prolly can tune this up later. But the npm install does work. That was the easiest path to get started with Yarn, and Firefox and Chrome running the the build. I'd like to revisit this setup at some point.

File: addon-test-support/helpers.js (hunk 6 above)
Comment (pixelhandler) [inline addon-test-support/helpers.js]: oops, can fix that.
